### PR TITLE
fix: add authorization token to generate code

### LINKED ISSUE #1791: Basic and Bearer Token is not exported with gerate code
When click the generate code button the Authorization headers with token (Bearer or Basic) is not exported

bruno-auth-token

### PR DESCRIPTION
**Pull request by leoferreiralima**
Fixes #1791

# Description

This PR add to Bruno the ability of generate code with Authorization headers of basic and bearer mode

bruno-auth-token-fix

### Contribution Checklist:

- [x] **The pull request only addresses one issue or adds one feature.**
- [x] **The pull request does not introduce any breaking changes**
- [x] **I have added screenshots or gifs to help explain the change if applicable.**
- [x] **I have read the [contribution guidelines](https://github.com/usebruno/bruno/blob/main/contributing.md).**
- [x] **Create an issue and link to the pull request.**

### CODE REVIEW

**Comment by leobenkel** [comment]
thank you. this feature is missing

File: packages/bruno-app/src/utils/codegenerator/auth.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import get from 'lodash/get';
+
+export const getAuthHeaders = (collectionRootAuth, requestAuth) => {
+  const auth = collectionRootAuth && ['inherit', 'none'].includes(requestAuth.mode) ? collectionRootAuth : requestAuth;
+
+  if (auth.mode === 'basic') {
```
Comment (varunvora) [inline packages/bruno-app/src/utils/codegenerator/auth.js]: nit: `switch` may be cleaner here. Similar to `packages/bruno-app/src/utils/codegenerator/har.js`

Comment (varunvora) [inline packages/bruno-app/src/utils/codegenerator/auth.js]: Any plans on supporting other auth methods

File: packages/bruno-app/src/components/Sidebar/Collections/Collection/CollectionItem/GenerateCodeItem/CodeView/index.js [hunk 2]
```diff
@@ -20,10 +21,16 @@ const CodeView = ({ language, item }) => {
...
-  const headers = [...(collection?.root?.request?.headers || []), ...(requestHeaders || [])];
+  const collectionRootAuth = collection?.root?.request?.auth;
+  const requestAuth = item.draft ? get(item, 'draft.request.auth') : get(item, 'request.auth');
 
-  let snippet = '';
+  const headers = [
+    ...(collection?.root?.request?.headers || []),
+    ...(requestHeaders || []),
+    ...getAuthHeaders(collectionRootAuth, requestAuth)
+  ];
```
Comment (varunvora) [inline packages/bruno-app/src/components/Sidebar/Collections/Collection/CollectionItem/GenerateCodeItem/CodeView/index.js]: The list of headers is translated to a map. Previously mapped items may be overridden. If headers with the key "Authorization" are declared at the request level, they will be overridden by collection-level auth.

**Review by varunvora — CHANGES_REQUESTED** [review]

**Comment by varunvora** [comment]
Thank you! I hope this gets merged soon.

File: packages/bruno-app/src/utils/codegenerator/auth.js (hunk 1 above)
Comment (leoferreiralima) [reply packages/bruno-app/src/utils/codegenerator/auth.js]: it's based on my expirence with insomnia, only basic and bearer works. But in the future we can add other methods

File: packages/bruno-app/src/components/Sidebar/Collections/Collection/CollectionItem/GenerateCodeItem/CodeView/index.js (hunk 2 above)
Comment (leoferreiralima) [reply packages/bruno-app/src/components/Sidebar/Collections/Collection/CollectionItem/GenerateCodeItem/CodeView/index.js]: thanks i will change this!

File: packages/bruno-app/src/utils/codegenerator/auth.js (hunk 1 above)
Comment (leoferreiralima) [reply packages/bruno-app/src/utils/codegenerator/auth.js]: done

**Comment by leoferreiralima** [comment]
@varunvora @sanjai0py changes are done, do you can review? to see if something is missing

**Comment by varunvora** [comment]
Looks good! Thanks so much for this. @sanjai0py can we have this merged soon?

**Review by varunvora — APPROVED** [review]

**Comment by hungdh0x5e** [comment]
+1
I really need this feature

**Comment by Moribund7** [comment]
@sanjai0py do you have any updates regarding this one? Could it be merged?

**Comment by Moribund7** [comment]
@varunvora do you know who from the maintainers could approve the workflow run?

**Comment by fritz-trawa** [comment]
[helloanoop](https://github.com/helloanoop) do you need help looking after this repo?

**Comment by Moribund7** [comment]
@leoferreiralima could you merge this one?

**Comment by Moribund7** [comment]
Or @varunvora do you have permissions to merge this one?

**Comment by leoferreiralima** [comment]
> @leoferreiralima could you merge this one?

i don't have permission to do that. We have to wait the maintainers merge it

File: packages/bruno-app/src/utils/codegenerator/auth.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import get from 'lodash/get';
+
+export const getAuthHeaders = (collectionRootAuth, requestAuth) => {
+  const auth = collectionRootAuth && ['inherit', 'none'].includes(requestAuth.mode) ? collectionRootAuth : requestAuth;
```
Comment (lohxt1) [inline packages/bruno-app/src/utils/codegenerator/auth.js:4]: you might want to remove 'none' from the array

**Comment by helloanoop** [comment]
Merged!

Thanks for working on this @leoferreiralima !
Appreciate everyone's patience on the delayed merge.

**Pull request merged**